sklearn/utils/testing.py

```diff
@@ -422,6 +441,172 @@ def assert_raise_message(exceptions, message, function, *args, **kwargs):
                              (names, function.__name__))
 
 
+def _assert_same_model_method(method, X, estimator1, estimator2, msg=None):
+    method_err = '%r\n\nhas %s, but\n\n%r\n\ndoes not.'
+    if hasattr(estimator1, method):
+        if not hasattr(estimator2, method):
+            raise AssertionError(method_err % (estimator1, method, estimator2))
+
+        # Check if the method(X) returns the same for both models.
+        res1 = getattr(estimator1, method)(X)
+        res2 = getattr(estimator2, method)(X)
+
+        if sp.issparse(res1) or sp.issparse(res2):
+            same_model = _assert_sparse_allclose(res1, res2)
```

Comment:
In that case, can you use `assert_allclose` instead of `np.allclose` below? Also, I think the current code doesn't raise the correct exception message if the sparse code path is being used.